### PR TITLE
Moving create backup of server vars

### PR DESCRIPTION
**Pull request by hkim823**
To after dump all vars to yaml (originally before)

### CODE REVIEW

**Comment by jarv** [comment]
:+1:

**Comment by hkim823** [comment]
Except now I realize that this defeats the purpose of the original PR (and is pointless after the dump all vars"

**Comment by hkim823** [comment]
New commit to hopefully fix if file doesn't exist first

File: playbooks/roles/edx_ansible/tasks/deploy.yml [hunk 1]
```diff
@@ -34,17 +34,21 @@
...
     dest={{ edx_ansible_var_file }}.old
-  when: EDX_ANSIBLE_DUMP_VARS
+  when: edx_ansible_var_file_exists or EDX_ANSIBLE_DUMP_VARS
 
 - name: dump all vars to yaml
   template: src=dumpall.yml.j2 dest={{ edx_ansible_var_file }} mode=0600
   when: EDX_ANSIBLE_DUMP_VARS
 
-- name: create symlink for config file
+ - name: create symlink for config file
```
Comment (jarv) [inline playbooks/roles/edx_ansible/tasks/deploy.yml]: Why is there a space here?

File: playbooks/roles/edx_ansible/tasks/deploy.yml [hunk 2]
```diff
@@ -34,17 +34,21 @@
...
+- name: check to see if server vars already exists
+  stat: {{ edx_ansible_var_file }}
+  register: edx_ansible_var_file_exists
+
 - name: create backup of server vars
   copy: >
     src={{ edx_ansible_var_file }}
     dest={{ edx_ansible_var_file }}.old
-  when: EDX_ANSIBLE_DUMP_VARS
+  when: edx_ansible_var_file_exists or EDX_ANSIBLE_DUMP_VARS
```
Comment (jarv) [inline playbooks/roles/edx_ansible/tasks/deploy.yml]: I think you need to check a property of this var, not the var itself

**Comment by hkim823** [comment]
squashed commits

**Comment by hkim823** [comment]
@feanil ready for review

File: playbooks/roles/edx_ansible/tasks/deploy.yml [hunk 3]
```diff
@@ -34,11 +34,15 @@
...
+- name: check to see if server vars already exists
+  stat: path={{ edx_ansible_var_file }}
+  register: edx_ansible_var_file_exists
+
 - name: create backup of server vars
   copy: >
     src={{ edx_ansible_var_file }}
     dest={{ edx_ansible_var_file }}.old
-  when: EDX_ANSIBLE_DUMP_VARS
+  when: edx_ansible_var_file_exists.stat.exists or EDX_ANSIBLE_DUMP_VARS
```
Comment (feanil) [inline playbooks/roles/edx_ansible/tasks/deploy.yml]: Why does this need the `or EDX_ANSIBLE_DUMP_VARS` it seems like it should be an `and` or not be there at all.

Comment (hkim823) [inline playbooks/roles/edx_ansible/tasks/deploy.yml]: That's is true, there isn't a scenario where we would dump the vars and the file wouldn't exist.  I'll remove it

**Comment by feanil** [comment]
:+1:

**Pull request merged**